src/coreclr/vm/typehandle.h

```diff
@@ -159,43 +159,16 @@ class TypeHandle
     }
 
         // Methods for probing exactly what kind of a type handle we have
-    FORCEINLINE BOOL IsNull() const {
+    FORCEINLINE BOOL IsNull() const
+    {
         LIMITED_METHOD_DAC_CONTRACT;
-#ifdef _PREFIX_
-        if (m_asTAddr == 0) {
-#ifndef DACCESS_COMPILE
-            PREFIX_ASSUME(m_asPtr == NULL);
-#endif
-            return true;
-        }
-        else {
-#ifndef DACCESS_COMPILE
-            PREFIX_ASSUME(m_asPtr != NULL);
-#endif
-            return false;
-        }
-#else
         return(m_asTAddr == 0);
-#endif
     }
 
-    // Note that this returns denormalized BOOL to help the compiler with optimizations
-    FORCEINLINE BOOL IsTypeDesc() const  {
+    FORCEINLINE BOOL IsTypeDesc() const
```

Comment:
```suggestion
    bool IsTypeDesc() const
```
No compiler should need FORCEINLINE hint for this.
Also, you can change it to `bool` while you are on it. (This type has a mix of BOOL and bool, so this is not introducing new inconsistency.)